meson.build

```diff
@@ -199,6 +199,33 @@ gumjs_extra_deps = []
 gumjs_extra_requires = []
 gumjs_extra_libraries = []
 
+glib_src = '''
+#include <glib.h>
+
+#ifdef GLIB_STATIC_COMPILATION
+# error GLib is static
+#endif
+'''
+
+has_shared_glib = cc.compiles(glib_src, name: 'compiling with shared GLib', dependencies: [glib_dep])
+have_gir = false
+
+if has_shared_glib
+  gir = find_program('g-ir-scanner')
+  have_gir = gir.found()
+endif
+
+if cc.has_function('g_thread_set_callbacks', dependencies: [glib_dep])
+  cdata.set('HAVE_FRIDA_GLIB', 1)
+else
+  elf_dep = dependency('libelf')
```

Comment:
These two lines should be moved into/consolidated with with the code inside:

`if host_os_family == 'linux' or host_os_family == 'qnx'`